lib/ai_helper/entry_point.rb

```diff
@@ -19,6 +19,20 @@ def inject_into(plugin)
             thread_id: thread.id,
           )
         end
+
+        plugin.add_to_serializer(
+          :current_user,
+          :ai_helper_prompts,
+          include_condition: -> do
+            SiteSetting.composer_ai_helper_enabled && scope.authenticated? &&
+              scope.user.in_any_groups?(SiteSetting.ai_helper_allowed_groups_map)
+          end,
+        ) do
+          ActiveModel::ArraySerializer.new(
+            DiscourseAi::AiHelper::Assistant.new.available_prompts,
```

Comment:
this is great for starters, but I think we can eliminate this query to the database with a cache, feels kind of important cause there is no need to run through all of this work on every page load for users with access.

Reply:
not blocking can happen in a followup pr.